media_management_api/media_auth/authentication.py

```diff
@@ -39,14 +39,15 @@ def authenticate_header(self, request):
     
     def authenticate(self, request):
         jwt = get_access_token_from_request(request, "Bearer ")
-
         if not jwt:
             return None
 
+        logger.debug(f"Attempting to authenticate jwt: {jwt}")
         decoded_token = decode_jwt(jwt)
         if not decoded_token:
-            return None
-        
+            raise exceptions.AuthenticationFailed("JWT authentication failed")
```

Comment:
Does the exception propagate to the parent function(not sure if i'm asking this correctly)?

Reply:
Yes, the exception will propagate up to the caller. This `CustomJWTAuthentication` class will be called by a django rest framework `APIView` (or a subclass of it) to authenticate the request, and the expected behavior is that the view will catch the exception and automatically return a JSON response with a 403 status code. 

There's some documentation here, although it doesn't go into the nitty gritty of how the exception propagates:
- https://www.django-rest-framework.org/api-guide/exceptions/#authenticationfailed
- https://www.django-rest-framework.org/api-guide/authentication/#custom-authentication